src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2019-12-30 20:52
+from __future__ import unicode_literals
+
+from datetime import timedelta, datetime
+
+from django.db import migrations
+
+from sentry import options
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Inserts Postgres events into the eventstream if there are recent events in Postgres.
+
+    This is for open source users migrating from 9.x who want to keep their events.
+    If there are no recent events in Postgres, skip the backfill.
+    """
+    from sentry import eventstore, eventstream
+    from sentry.utils.query import RangeQuerySetWrapper
+
+    Event = apps.get_model('sentry', 'Event')
+    Group = apps.get_model('sentry', 'Group')
+    Project = apps.get_model('sentry', 'Project')
+
+    # Use 90 day retention if the option has not been set or set to 0
+    DEFAULT_RETENTION = 90
+
+    retention_days = options.get("system.event-retention-days") or DEFAULT_RETENTION
```

Comment:
Note, this allows users to set a value > 90 days as long as it is not infinity. @BYK I have no opinion on whether we'd prefer to enforce 90 days as a max - thoughts?

Reply:
I think this is the best for now, good call!